includes/dintero-checkout-functions.php

```diff
@@ -215,9 +223,15 @@ function dintero_confirm_order( $order, $transaction_id ) {
 
 	// Save shipping id to the order.
 	$shipping = $order->get_shipping_methods();
-	if ( ! empty( $shipping ) ) {
-		$shipping_option_id = $dintero_order['shipping_option']['id'] ?? reset( $shipping );
-		$order->update_meta_data( '_wc_dintero_shipping_id', $shipping_option_id );
+	if ( ! empty( $shipping ) && empty( $order->get_meta( '_wc_dintero_shipping_id' ) ) ) {
```

Comment:
Just to verify here, should we not update it if its already set? What if its been changed in some way potentially? Im mainly thinking about cases where a subscription has the payment method updated when for example a card has expired, could that not cause a new shipping id to be set?

Reply:
Yes, this one is not supposed to be modified once set. I believe this was related to an issue when renewing the subscription, where the ID of the shipping option was modified, although it is the same shipping method. Dintero doesn't allow you to modify the line ID, which is what this ID is used for, once the order has been created.